File: frontend/src/lib/applicationReview.ts
import { api } from '@/lib/api';

/**
 * Phase 7.12 — AI Review Pipeline API client. Thin typed wrappers over the additive
 * /api/application-review/* surface. Dark by default: reads 404 until the pipeline is enabled, so
 * callers must tolerate absent data (never fabricate a review client-side). POST /run reviews only —
 * it never submits an application.
 */

export type ReviewVerdict = 'READY' | 'HUMAN_REVIEW' | 'BLOCKED';
export type ConsistencyStatus = 'PASS' | 'WARNING' | 'FAIL';
export type QualityCategory = 'EXCELLENT' | 'STRONG' | 'GOOD' | 'WEAK' | 'BLOCKED';

export interface ApplicationReviewResponse {
  id: string;
  applicationPackageId: string;
  userId: string;
  jobId: string;
  packageVersion: number;
  resumeScore: number | null;
  atsScore: number | null;
  companyFitScore: number | null;
  learningConfidence: number | null;
  consistencyStatus: ConsistencyStatus | null;
  qualityScore: number | null;
  qualityCategory: QualityCategory | null;
  verdict: ReviewVerdict;
  confidence: number | null;
  reasons: string | null;
  correlationId: string | null;
  reviewVersion: number;
  createdAt: string;
  updatedAt: string;
}

export interface ReviewHistoryResponse {
  id: string;
  applicationReviewId: string;
  packageVersion: number;
  reviewVersion: number;
  qualityScore: number | null;
  qualityCategory: QualityCategory | null;
  verdict: ReviewVerdict;
  confidence: number | null;
  consistencyStatus: ConsistencyStatus | null;
  createdAt: string;
}

export interface ReviewQualityResponse {
  applicationPackageId: string;
  qualityScore: number | null;
  qualityCategory: QualityCategory | null;
  verdict: ReviewVerdict;
  confidence: number | null;
}

export async function getReview(packageId: string) {
  return (await api.get<ApplicationReviewResponse>(`/api/application-review/${packageId}`)).data;
}

export async function getReviewHistory(packageId: string) {
  return (await api.get<ReviewHistoryResponse[]>(`/api/application-review/${packageId}/history`)).data;
}

export async function getReviewQuality(packageId: string) {
  return (await api.get<ReviewQualityResponse>(`/api/application-review/${packageId}/quality`)).data;
}

/** Run the review pipeline over a package. Review only — never submits. */
export async function runReview(packageId: string) {
  return (await api.post<ApplicationReviewResponse>(`/api/application-review/${packageId}/run`)).data;
}

/** One parsed reviewer entry from the review's reasons JSON. */
export interface ReviewerReason {
  reviewer: string;
  score?: number | null;
  status?: string;
  category?: string;
  reasons?: string[];
}

export function parseReasons(reasons: string | null | undefined): ReviewerReason[] {
  if (!reasons) return [];
  try {
    return JSON.parse(reasons) as ReviewerReason[];
  } catch {
    return [];
  }
}

File: backend/src/main/java/ai/careerpilot/domain/ApplicationReview.java
package ai.careerpilot.domain;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.Instant;
import java.util.UUID;

/**
 * Phase 7.12 — the HEAD AI review per Application Package: the final quality gate's verdict plus each
 * reviewer's score. It REVIEWS the Phase 7.11 package (resume/ATS/recommendation/company/learning
 * assessments) and never mutates any of them. {@link ApplicationReviewHistory} keeps the immutable
 * append-only trail of every review run.
 */
@Entity
@Table(name = "application_review")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class ApplicationReview {

    @Id @GeneratedValue
    private UUID id;

    @Column(name = "application_package_id", nullable = false) private UUID applicationPackageId;
    @Column(name = "user_id", nullable = false) private UUID userId;
    @Column(name = "job_id", nullable = false) private UUID jobId;
    @Column(name = "package_version", nullable = false) private Integer packageVersion;

    @Column(name = "resume_score") private Integer resumeScore;
    @Column(name = "ats_score") private Integer atsScore;
    @Column(name = "company_fit_score") private Integer companyFitScore;
    @Column(name = "learning_confidence") private Integer learningConfidence;
    @Column(name = "consistency_status") private String consistencyStatus; // PASS | WARNING | FAIL
    @Column(name = "quality_score") private Integer qualityScore;          // 0-100
    @Column(name = "quality_category") private String qualityCategory;     // EXCELLENT | STRONG | GOOD | WEAK | BLOCKED

    @Column(nullable = false) private String verdict; // READY | HUMAN_REVIEW | BLOCKED
    @Column private Integer confidence;               // 0-100
    @Column(columnDefinition = "text") private String reasons;
    @Column(name = "correlation_id") private UUID correlationId;
    @Column(name = "review_version", nullable = false) private Integer reviewVersion;

    @CreationTimestamp @Column(name = "created_at", updatable = false) private Instant createdAt;
    @UpdateTimestamp @Column(name = "updated_at") private Instant updatedAt;
}
